src/firmware_new/boards/frankie_v1/io/proto_multicast_communication_profile.h

```diff
@@ -0,0 +1,172 @@
+ *
+ * @param profile The profile to get the fields from
+ *
+ * @return the void ptr to the protobuf_struct
+ */
+const pb_field_t* io_proto_multicast_communication_profile_getProtoFields(
+    ProtoMulticastCommunicationProfile_t* profile);
+
+/**
+ * Get the maximum protobuf size
+ *
+ * @param profile The profile to get the value from
+ *
+ * @return the maximum size of the protobuf
+ */
+uint16_t io_proto_multicast_communication_profile_getMaxProtoSize(
+    ProtoMulticastCommunicationProfile_t* profile);
+
+/**
+ * Locks the profile to implicitly lock the internally tracked pointer
+ * to the protobuf_struct. The respective profile lock must be acquired before
+ * reading/writing the values inside the proto struct.
+ *
+ * NOTE: call blocks until the lock is acquired. If there is a higher
+ * priority task waiting for the mutex, the task holding this lock will
+ * inherit the waiting tasks priority. (osMutexPrioInherit)
+ *
+ * @param communication_profile The profile to lock
+ */
+void io_proto_multicast_communication_profile_acquireLock(
+    ProtoMulticastCommunicationProfile_t* profile);
+
+/**
+ * Release the profile lock.
+ *
+ * @pre lock must be acquired through the call to "communication_profile_acquireLock"
+ *
+ * @param communication_profile The profile to release the mutex for
+ */
+void io_proto_multicast_communication_profile_releaseLock(
+    ProtoMulticastCommunicationProfile_t* profile);
+
+/**
+ * Signal the tasks waiting on the communication_profile with the provided events.
+ *
+ * @pre do NOT hold the profile lock, this will cause everything to grind to a halt
+ *
+ * @param communication_profile The profile that the notification will be "sent on". Any
+ * task blocking with io_proto_multicast_communication_profile_blockUntilEvents will be
+ * notified.
+ * @param events The events to notify, bitwise or of
+ * ProtoMulticastCommunicationEventFlags_t
+ */
+void io_proto_multicast_communication_profile_notifyEvents(
+    ProtoMulticastCommunicationProfile_t* profile, uint32_t events);
+
+/**
+ * Block the calling function until a new protobuf value has been received over the
+ * network.
+ *
+ * @pre do NOT hold the profile lock, this will cause everything to grind to a halt
+ *
+ * @param communication_profile The profile that the notification will occur on
+ * @param events The events to block on, bitwise or of
+ * ProtoMulticastCommunicationEventFlags_t
+ *
+ * @return The event flags that were notified causing this function to return.
+ *         bitwise OR of ProtoMulticastCommunicationEventFlags
+ */
+uint32_t io_proto_multicast_communication_profile_blockUntilEvents(
+    ProtoMulticastCommunicationProfile_t* profile, uint32_t events);
+
+/**
+ * Destroy the given ProtoMulticastCommunicationProfile_t
+ *
+ * @param communication_profile The profile to delete
+ */
+void io_proto_multicast_communication_profile_destroy(
+    ProtoMulticastCommunicationProfile_t* profile);
```

Comment:
nit; for consistency can we put this directly after `create`?

Reply:
moved